### DISCUSSION TITLE
Migrating Monorepo + Git Submodule setup to v3

### DISCUSSION THREAD

**Discussion by sameera** [chakra-ui/chakra-ui#9042] [General]
I have a shared UI library that I created for all my apps that sits in its own Git repo. List call it UiLib.
I use NX in my application projects and in each of these projects, add UiLib as a git submodules at /libs/shared/UiLib.
UiLib is still evolving and therefore, it's not ready yet to be packaged into a NPM module; hence the submodule approach.

This worked well enough in v2. I just started a new project and tried migrating to v3 and am stuck with the snippets.
For starters, `npx @chakra-ui/cli snippet add` is not compatible with monorepo structures. It seems to assume that it's in a single project repo.

How can I adjust my approach to v3?
Would it make sense to create a new repo for snippets and  mount that as a submodule as well?

**Comment by segunadebayo** [comment]
Run the CLI from the package that owns the UI, and pass `--outdir`.

```bash
npx @chakra-ui/cli snippet add --outdir libs/shared/UiLib/src/components/ui
```
